### PR TITLE
Card Container Component

### LINKED ISSUE #26: CardContainer
## Common Component - CardContainer

 we need a `Card Container` to render cards of providers.

- props: array of providers.
- [ ] It will be used in Pages :
          landing page.

- [ ] UI Image From Design :+1: 
Screenshot from 2021-03-24 19-22-38

### PR DESCRIPTION
**Pull request by Osama-you**
- Update card style
- Make card Container 
- Add Pagination
and the code example:
```jsx
<CardContainer
    provides={[
      {
        id: 12,
        username: 'Dugald',
        avatar:
          'https://cdn0.iconfinder.com/data/icons/male-profession-flat/32/engineer-512.png',
        location: '76 Birchwood Trail',
        bio:
          'Nullam porttitor lacus at turpis. Donec posuere metus vitae ipsum. Aliquam non mauris.\n\nMorbi non lectus. Aliquam sit amet diam in magna bibendum imperdiet. Nullam orci pede, venenatis non, sodales sed, tincidunt eu, felis.',
        price_hour: 7.8,
        rating: 3,
        cover_image:
          'https://zos.alipayobjects.com/rmsportal/jkjgkEfvpUPVyRjUImniVslZfWPnJuuZ.png',
        title: 'posuere metus vitae ',
      },
    ]}
    title="Nullam porttitor"
  />
```
- and I fixed avatar component

### CODE REVIEW

**Review by mAbadsa — DISMISSED** [review]

**Review by Ahmed-aed — DISMISSED** [review]

**Review by Ahmed-aed — DISMISSED** [review]

File: client/src/Components/CardContainer/index.js [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+  const numEachPage = 8;
+  const [limit, setLimit] = useState([0, numEachPage]);
+
+  return (
+    <div className="container" {...rest}>
+      <Title id="container-title" level={2}>
+        {title}
+      </Title>
+      <div className="card-container">
+        {data && data.length !== 0 ? (
```
Comment (mariamisa) [inline client/src/Components/CardContainer/index.js]: rename the data prop name to providers => for more descriptive

File: client/src/Components/CardContainer/index.js [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+                key={item.id}
+                id={item.id}
+                Name={item.username}
+                priceByHour={item.price_hour}
+                ImageSrc={item.cover_image}
+                avatarImage={item.avatar}
+                TitleJob={item.title}
+                city={item.location}
+                rate={item.rating}
+                descriptions={item.bio}
```
Comment (mariamisa) [inline client/src/Components/CardContainer/index.js]: why you passing the values as single values, passing all as an object
```
<Card key={item.id}  provider={item} />
```

**Review by mariamisa — CHANGES_REQUESTED** [review]

File: client/src/Components/CardContainer/index.js [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+  return (
+    <div className="container" {...rest}>
+      <Title id="container-title" level={2}>
+        {title}
+      </Title>
+      <div className="card-container">
+        {data && data.length !== 0 ? (
+          data
+            .slice(limit[0], limit[1])
+            .map((item) => (
```
Comment (mAbadsa) [inline client/src/Components/CardContainer/index.js]: Use destructuring.

**Review by mAbadsa — CHANGES_REQUESTED** [review]

**Review by zeinjendeya — DISMISSED** [review]

File: client/src/Components/CardContainer/index.js (hunk 1 above)
Comment (Osama-you) [reply client/src/Components/CardContainer/index.js]: done

File: client/src/Components/CardContainer/index.js (hunk 2 above)
Comment (Osama-you) [reply client/src/Components/CardContainer/index.js]: done

File: client/src/Components/CardContainer/index.js (hunk 3 above)
Comment (Osama-you) [reply client/src/Components/CardContainer/index.js]: not needed anymore.

**Review by mariamisa — APPROVED** [review]

**Pull request merged**